src/goto-programs/abstract-interpretation/interval_analysis.cpp

```diff
@@ -39,18 +39,81 @@ static inline void simplify_guard(expr2tc &expr, const interval_domaint &state)
   simplify(expr);
 }
 
+static void optimize_expression(expr2tc &expr, const interval_domaint &state)
+{
+  ///////////////////
+  // PRECONDITIONS //
+  ///////////////////
+  if(is_nil_expr(expr))
+    return;
+
+  // We can't simplify addr-of sub-expr
+  if(is_address_of2t(expr))
+    return;
+
+  // We can't replace the LHS of an assignment
```

Comment:
Seems somewhat clear, but the comment isn't really helpful in explaining anything.

Reply:
Many thanks, that makes it much clearer for readers who are not familiar with this code!